mmtrack/datasets/pipelines/transforms.py

```diff
@@ -134,13 +137,268 @@ def __call__(self, results):
+
+        # 1.3 pad image
+        img_crop_padded = cv2.copyMakeBorder(img_crop, y1_pad, y2_pad, x1_pad,
+                                             x2_pad, cv2.BORDER_CONSTANT)
+        # 1.4 generate padding mask
+        img_h, img_w, _ = img_crop_padded.shape
+        pdding_mask = np.ones((img_h, img_w))
+        end_x, end_y = -x2_pad, -y2_pad
+        if y2_pad == 0:
+            end_y = None
+        if x2_pad == 0:
+            end_x = None
+        pdding_mask[y1_pad:end_y, x1_pad:end_x] = 0
+
+        # 2. Resize image and padding mask
+        resize_factor = output_size / crop_size
+        img_crop_padded = cv2.resize(img_crop_padded,
+                                     (output_size, output_size))
+        pdding_mask = cv2.resize(pdding_mask,
+                                 (output_size, output_size)).astype(np.bool_)
+
+        return img_crop_padded, resize_factor, pdding_mask
+
+    def generate_box(self,
+                     bbox_gt,
+                     bbox_cropped,
+                     resize_factor,
+                     output_size,
+                     normalize=False):
+        """Transform the box coordinates from the original image coordinates to
+        the coordinates of the cropped image.
+
+        Args:
+            bbox_gt (ndarray): of shape (4, ) in [x1, y1, x2, y2] format.
+            bbox_cropped (ndarray): of shape (4, ) in [x1, y1, x2, y2] format.
+            resize_factor (float): the ratio of original image scale to cropped
+                image scale.
+            output_size (float): the size of output image.
+            normalize (bool): whether to normalize the output box.
+                Default to True.
+
+        Returns:
+            ndarray: generated box of shape (4, ) in [x1, y1, x2, y2] format.
+        """
+        assert output_size > 0
+        bbox_gt_center = (bbox_gt[0:2] + bbox_gt[2:4]) * 0.5
+        bbox_cropped_center = (bbox_cropped[0:2] + bbox_cropped[2:4]) * 0.5
+
+        bbox_out_center = (output_size - 1) / 2. + (
+            bbox_gt_center - bbox_cropped_center) * resize_factor
+        bbox_out_wh = (bbox_gt[2:4] - bbox_gt[0:2]) * resize_factor
+        bbox_out = np.concatenate((bbox_out_center - 0.5 * bbox_out_wh,
+                                   bbox_out_center + 0.5 * bbox_out_wh),
+                                  axis=-1)
+
+        return bbox_out / output_size if normalize else bbox_out
+
+    def __call__(self, results):
+        """Call function. For each dict in results, crop image like Stark did.
+
+        Args:
+            results (list[dict]): list of dict from
+                :obj:`mmtrack.base_sot_dataset`.
+
+        Returns:
+            List[dict]: list of dict that contains cropped image and
+                the corresponding groundtruth bbox.
+        """
+        outs = []
+        for i, _results in enumerate(results):
+            image = _results['img']
+            gt_bbox = _results[_results.get('bbox_fields', [])[0]][0]
```

Comment:
can we change all `_results.get('bbox_fields', [])[0]` to `'gt_bboxes'`?

Reply:
I think it's no problem.

Reply:
Then please also change the codes in `SeqCropLikeSiamFC `